### ISSUE TITLE
Examples needed

### ISSUE THREAD

**Issue by nicola** [marcelklehr/gulf#24]
I have been playing around with the last few hours.
I all looks really great and the code is really readable.

However it is non-trivial to understand how to use it (overall),
it would be great if there would be some examples describing:

- [ ] Setting up gulf with websocket/http (which I guess can be done by using Link)
- [ ] Setting up a very simple html editor

**Comment by marcelklehr** [comment]
Check out https://github.com/marcelklehr/warp ;) I'm updating it with the latest versions of the libraries atm.

**Comment by marcelklehr** [comment]
I'm struggling to find time for this, sadly. As part of the refactoring of Hive.js, I've also opted to refactor gulf. @derhuerst How far have you come with gulf? Once the refactoring of gulf is done, do you think you could help write thorough docs?

**Comment by derhuerst** [comment]
@marcelklehr maybe. I don't have so much time either unfortunately (or rather dozens of other projects). But let me know this is done. I will have a look at what i built using gulf tonight.

**Comment by derhuerst** [comment]
I created [`gulf-editor-widget`](https://github.com/derhuerst/gulf-editor-widget), which should work theoretically, but with the following server & client, I can't get it to work.

```js
# server
const gulf = require('gulf')
const textOT = require('ot-text').type
const net = require('net')

const welcome = textOT.create('Hello World!')

gulf.Document.create(new gulf.MemoryAdapter(), textOT, welcome, (err, doc) => {
	if (err) throw err

	const server = net.createServer((socket) => {
		const follower = doc.slaveLink()
		socket.pipe(follower)
		follower.pipe(socket)
	})
	server.listen(7777, (err) => {
		if (err) throw err
		console.info('Listening on port 7777.')
	})
})
```

```js
# client
'use strict'

const blessed = require('blessed')
const Editor = require('editor-widget')
const gulf = require('gulf')
const textOT = require('ot-text').type
const net = require('net')

const connect = require('./gulf-editor-widget')

console.log = console.warn = console.info = () => {}



const screen = blessed.screen({smartCSR: true})
screen.key(['escape', 'C-d'], () => process.exit(0))
screen.key(['escape', 'C-c'], () => process.exit(1))

const editor = new Editor({
	parent: screen,
	top: 0, left: 0,
	width: '100%', height: '100%',
	gutter: {width: 0, hidden: true}
})
editor.language('js')



const doc = new gulf.EditableDocument(new gulf.MemoryAdapter(), textOT)
connect(doc, editor)

const socket = net.connect(7777, () => {
	const leader = doc.masterLink()
	leader.pipe(socket)
	socket.pipe(leader)
})
```

It's a bit tricky that [`blessed`](https://github.com/chjj/blessed) swallows both `stdout` & `stderr`. I will have a look tomorrow.

**Comment by marcelklehr** [comment]
Cool stuff! I've just released a fix for gulf that should make it work. Messages weren't correctly separated on reception.

**Comment by derhuerst** [comment]
Still doesn't work. Shall I post some logs?

**Comment by marcelklehr** [comment]
Yes, in a new issue, please :)
